src/components/OptionsSelector.js

```diff
@@ -207,12 +214,12 @@ class OptionsSelector extends Component {
                     ref={el => this.list = el}
                     optionHoveredStyle={styles.hoveredComponentBG}
                     onSelectRow={this.selectRow}
-                    sections={this.props.sections}
+                    sections={this.state.sections || this.props.sections}
                     focusedIndex={this.state.focusedIndex}
                     selectedOptions={this.props.selectedOptions}
                     canSelectMultipleOptions={this.props.canSelectMultipleOptions}
                     hideSectionHeaders={this.props.hideSectionHeaders}
-                    headerMessage={this.props.headerMessage}
+                    headerMessage={headerMessage}
```

Comment:
don't need a separate variable for this, let's just put `this.props.getCustomHeaderMessage(this.searchValue)` here